### ISSUE TITLE
Add porticoes as type of sidewalk

### ISSUE THREAD

**Issue by fabiobettani** [streetcomplete/StreetComplete#2452]
When the user is asked whether a given road has a sidewalk on a given side, they are only given two options: "no sidewalk" or "sidewalk".

In some countries, and especially in my home country (Italy), a common feature of city centres are porticoes. See for example: http://shorturl.at/gsEH7
In this case, the question is misleading: a road like this certainly has no sidewalk, but the correct syntax on OSM would actually be sidewalk:both, sidewalk:both:covered=arcade. As my hometown is filled with porticoes, up to now I have been ignoring all sidewalk quests, exactly because I can't decide whether the answer should be "yes" or "no" in such cases.

I suggest that, at least in selected countries such as Italy, more alternatives are added, other than "no sidewalk" or "sidewalk": these should be "arcade" and "colonnade" (see https://wiki.openstreetmap.org/wiki/Key:covered).

![Screenshot_20210103-141841](https://user-images.githubusercontent.com/66837227/103479587-0c9b3780-4dcf-11eb-8be3-2c3be53a1744.jpg)

**Comment by matkoniecz** [comment]
It is fairly rare in Poland but it is clearly sidewalk to me, just covered.

Maybe Italian translation can be improved? (or maybe I am misinterpreting words).

**Comment by westnordost** [comment]
What would be your suggestion for a wording for "yes, sidewalk!" that includes arcades and collonades?

**Comment by fabiobettani** [comment]
In Italian, the word for "sidewalk" clearly refers to the common kind of sidewalk that we would expect on a modern road: an unsheltered walkway, usually separated by a curb. That's why I never thought of arcades and colonnades as a subset of sidewalks.

English is not my native language and I'm still not sure what a "sidewalk" is, to a native speaker. The Wikipedia entry for "sidewalk" does not explicitly rule out arcades and colonnades, but doesn't mention them, either. The Wikipedia entry for "arcade" defines them as a form of "sheltered walkway for pedestrians" (and not "a sidewalk").

I'm suggesting to give the user a choice between four alternatives: "no sidewalk", "unsheltered sidewalk", "arcade" and "colonnade".

If we choose to restrict the answer to only two options, the question could be phrased along the lines of "Does this street have a walkway?", and the answers could be "No walkway" and "Walkway". And we would possibly need a separate quest in order to mark such walkways as either uncovered walkways, arcades or colonnades.

**Comment by westnordost** [comment]
Nobody would know what is meant by a "walkway". Maybe that would work in Italian? Then you can change the Italian translation.

**Comment by fabiobettani** [comment]
In Italian I can find no single word to describe sidewalks, arcades and colonnades, so I would use an expression like "percorso pedonale" (literally "pedestrian path"), and I would add a note in POeditor clarifying to future translators why we should not be using the usual Italian translation of "sidewalk".

But that leaves my doubt open: is it clear to everyone that the current wording ("sidewalk" or its current translation in other languages) explicitly includes arcades and colonnades? Any feedback from native speakers of other languages is welcome.

If we decide that tweaking the Italian translation is enough, I would then formulate a separate proposal for a new quest, limited to Italy and eventually other selected countries, which, under the current wording, would sound like "What kind of sidewalk does this road have?", with the possible options "Unsheltered sidewalk", "Arcade" and "Colonnade".

**Comment by westnordost** [comment]
In German, if there is a road that has no space for pedestrians other than an arcade, I'd consider the arcade as a "covered sidewalk". But in any case, if the user is unsure about whether an colonnade counts as a sidewalk or not, it doesn't hurt to leave a note because that opens the possibility that other mappers will also add `covered=yes`, which is even better than just tagging that there is a sidewalk.

**Comment by westnordost** [comment]
Here is one example in Hamburg. The cycleway is not covered, the pedestrian part of the sidewalk is covered:
https://www.google.de/maps/@53.55169,9.9856625,3a,75y,205.33h,76.74t/data=!3m6!1e1!3m4!1sFRTJ_eU4ya9RtO_8FNI54g!2e0!7i13312!8i6656

The colonnades in this road (the name of the road is "Colonnaden" by the way ;-) ) I would not consider as a sidewalk because the whole road is a pedestrian zone already and there is no curb:
https://www.google.de/maps/@53.5562,9.9905,3a,75y,191.38h,86.86t/data=!3m8!1e1!3m6!1sAF1QipNLflFTczyvMM7h-NhHinKxEkY6b8OsYNrFW-8v!2e10!3e11!6shttps:%2F%2Flh5.googleusercontent.com%2Fp%2FAF1QipNLflFTczyvMM7h-NhHinKxEkY6b8OsYNrFW-8v%3Dw203-h100-k-no-pi-0-ya295.73386-ro-0-fo100!7i5656!8i2828

**Comment by westnordost** [comment]
Anyway, I'll close this. You can refine the Italian translation if you think that helps

**Comment by westnordost** [comment]
According to the comments in POEditor, there seems to be the need to discuss this change. Since the discussion that led to the decision to change the translation happened here, let's keep the discussion in one place and continue here.

**Comment by Mannivu** [comment]
I think that adding porticoes to this quest is wrong, since porticoes are not part of the road (unlike a sidewalk) and have to be mapped with a dedicated way, since it must have tags that are not suitable for the main road ([covered=colonnade](https://wiki.openstreetmap.org/wiki/Tag:covered%3Dcolonnade)). Moreover, an Italian user would never think of a portico when asked if a street has a "marciapiede" (the Italian translation of "sidewalk").

**Comment by westnordost** [comment]
Well, as you know, there are two options how to map a sidewalk in OSM: as `sidewalk=*` on the road way or as a `highway=footway + footway=sidewalk` as a separate way. Both methods have some advantages, but the latter is certainly the one that allows more detail to be added, such as `covered=*` (well, theoretically `sidewalk:both:covered=*` would work too, but it gets complex quickly). In any case, at the time of writing, neither of these mapping schemes is deprecated in any way. Some sidewalks in Bologna might be tagged directly on the road way, some may be mapped as separate ways.

Before StreetComplete asks about whether a sidewalk exists or not, it makes sure that there is no separately mapped footway that could be a separately mapped sidewalk nearby. In other words, whenever the app asks about a sidewalk, there is guaranteed no sidewalk of any form recorded yet. Even if it is preferred by community consensus in Bologna to map such ways as separate ways, adding the sidewalk information on the road is still a positive contribution because it adds data. So when you say, it should be mapped separately, it absolutely can, later.

Since these two mapping schemes are interchangable, it means that the OSM definition of a _"sidewalk"_ is something like a _"footway along a carriageway that belongs conceptually to the road itself"_. I.e., if you use a pedestrian route planner, you would expect it to say "follow Via de Porto, then turn left at ...". Thus, the reasoning that I / @fabiobettani employed was that if the common translation for sidewalk doesn't cover all such "footway along a road" in Italian, the translation should be changed to accomodate that.

Another option too would be to choose a translation that is ambiguous enough that will make users doubt enough that they will never answer **no sidewalk** in that situation (because it would be wrong) but leave a note instead, describing that there is a porticoe and not a sidewalk. If that is possible.

**Comment by westnordost** [comment]
ping

**Comment by westnordost** [comment]
Okay, I am going to close this then.

**Comment by Mannivu** [comment]
We have discussed this on the Italian mailing list ([thread](http://gis.19327.n8.nabble.com/i-portici-sono-sidewalk-o-anche-che-cos-e-un-sidewalk-tp5982682.html)). The general thought is that porticoes are not a sidewalk (since there are ways with both porticoes and sidewalks), so the translation should not consider porticoes as a "yes" when the user is asked if a way has sidewalk or not. That said, the translation should not mention porticoes, but only "marciapiede".

**Comment by fabiobettani** [comment]
Actually, the discussion on the Italian mailing list reached no consensus at all. Maybe we need a ping there, if you feel like we need to go further.

Some objections were made, to which I replied with counter-argoments, and the discussion then dropped.

No example was given as why, in general terms, a portico that meets the conditions listed at https://wiki.openstreetmap.org/wiki/Sidewalks should not be treated as a sidewalk.

**Comment by Mannivu** [comment]
You changed the preexisting translation because you (without any discussion with the rest of the Italian community) thought that porticoes are a type of sidewalk and I think this is the wrong way of doing while participating in a communitarian project. Moreover, the general consensus in the ML is that porticoes are not sidewalks because they are not part of a road, they are external of the road bed and can't be treated as sidewalk i osm database.

**Comment by fabiobettani** [comment]
Actually, I changed the translation because consensus had been reached in this thread. Then, as soon as an objection was made, I opened myself a new thread on the Italian ML, in order to highlight this thread and get a broader consensus there. I couldn't think of a more communitarian way to do this. But let's set this aside, as we're here to focus on porticos :-)

> the general consensus in the ML is that porticoes are not sidewalks because they are not part of a road

To which I replied citing the exact paragraph in the Italian traffic law that defines roads, apparently including porticoes. I also mentioned real cases of the road police issuing fines against people who cycled along a portico - which shows that even the road police considers it as a form of sidewalk.

> they are external of the road bed

How is this relevant? https://wiki.openstreetmap.org/wiki/Sidewalks explicitly says that a sidewalk "may be at some distance from the road (but still associated with it)". And "it also may be separated from the road by some form of barrier".

> and can't be treated as sidewalk i osm database.

I probably don't get this one: to me, this actually sounds like what you are trying to prove, and not like a reason to support it. Correct me if I get your words wrong.

**Comment by Mannivu** [comment]
> onsensus had been reached in this thread
This thread? This issue in github isn't asking the italian community. This is not how you should operate: first you look consensus/opinions in the local community, then you try to see if the change in the original app/translation can be done. You can just say "two people from different communities" is a consensus!

In the ML there's no consensus that porticoes are sidewalk and you are not acknowledging that. You are the only one that says that porticoes are sidewalks while all the other participants are not sure about the definition of porticoes as sidewalks.

> How is this relevant?
This is relevant, because a sidewalk can be mapped simply with `sidewalk=yes` in the database, while a distant footway should be mapped with a dedicated way tagged with `highway=footway`.

>this actually sounds like what you are trying to prove
I'm not trying to prove it, I'm just trying to make you understand that you can't make a change and then search consensus in the proper community.

**Comment by westnordost** [comment]
I've read the last replies here in this thread. I agree with @Mannivu that one cannot simply change the definition of an existing tag. Even if consensus is reached, in fact, it would be faulty, because if the definition of a tag is changed, it then means that any data recorded under that tag before becomes (potentially) wrong or imprecise. But I do believe that @fabiobettani did not mean to change the definition but rather translate the definition for sidewalks in OSM how he understands it from the definition he read on the wiki to understandable common Italian, that's it.

So, the question is rather not what `sidewalk=*` should be, but what it is, i.e. where the `sidewalk=*` tag is used and how it is interpreted. And this is where you two disagree.

I've read a translation of that thread on the mailing list.

I must say I am a bit... disappointedthat the discussion on the ML revolves mostly around the legal definition of what is a sidewalk. This is but one aspect of it, and I must say, a rather irrelevant one. Who cares if for example in traffic law §14.3 (2) it is written that a sidewalk is defined to be at least 2.4m broad, has a kerb of at least 5cm height and must be on public property (fictional)? 
There are more important things to consider. They have been mentioned in this thread here and there already:
1. how is it currently defined in the wiki? How has it been defined historically in the wiki if it was different once?
2. have there been any discussions about it in the past? If yes, with which outcome?
3. how is it currently tagged?
4. how and for what purpose is the data used/interpreted by data consumers?

One reason why we have comminication channels with large audiences and not everybody decides for himself what is best is the hope that the more people listen in, the more likely all important sides of a matter are examined. That's why I wrote at the beginning that I am a little disappointed.

Especially point 3 and 4 seem to always fall short in (mailing list) discussions.

Pedestrian and wheelchair routers (f.e. OSRM) for example just want to know if pedestrians can walk alongside a road without walking on the carriageway. That's the definition of a sidewalk for routers. It doesn't matter if that sidewalk is covered, it does not matter if there is a kerb, it does not matter if there is a second sidewalk-like footway just next to it, it does not matter if it is not on public property as long as it is publicly accessible. The only thing that matters is that people can use it to traverse down a road, that's it.

Does this definition sound familiar? Yeah, that's your run-off-the-mill `highway=footway`. So I think there is no argument whether a porticoe can be tagged as `highway=footway`, of course it can. But just as mentioned in https://github.com/streetcomplete/StreetComplete/issues/2452#issuecomment-759457948 , there are two possible tagging schemas to tag "footways along the road", the other is to tag the sidewalk directly on the road way itself.
No matter which way is preferred, and I already said this in https://github.com/streetcomplete/StreetComplete/issues/2452#issuecomment-759457948 in bold letters, the alternative is that a road with only a porticoe and no other sidewalk and no separately mapped footway (yet) will be tagged with `sidewalk=no`. The result of that is that pedestrian routers will avoid this street because they (rightfully) think that people need to walk on the street and that's dangerous at a speed limit of f.e. 50 km/h.

In other words, it is a choice between clearly wrong data ("people must walk on the carriageway, maybe walking this road at all is forbidden") and data that depending on your personal definition of what is a sidewalk may be a little imprecise.
